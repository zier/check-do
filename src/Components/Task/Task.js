import React from 'react';
import PropTypes from 'prop-types';

import { Card, Tag, Icon  } from 'antd';

import './Task.css'

const Task = ({ id, title, description, isDone, tags, onClickToggleDone, onClickDelete }) => {
  // region Event: Buttons
  const onClickToggleTask = () => {
    onClickToggleDone(id)
  }

  const onClickDeleteTask = () => {
    onClickDelete(id)
  }
  // endregion

  let textLineThrough = { textDecoration: 'none' }
  if (isDone) {
    textLineThrough.textDecoration = 'line-through';
  }

  return (
    <Card style={{ padding: '0px' }} bordered={true}>
      <div className="control-buttons">
        <Icon onClick={onClickToggleTask} className="checkButton" type={isDone ? 'check-circle' : 'check-circle-o'} />
        <Icon onClick={onClickDeleteTask} className="closeButton" type="close" />
      </div>
      <h4 style={textLineThrough}>{title}</h4>
      <p style={textLineThrough}>{description}</p>
      {tags.map(tag => <Tag key={tag} className="taskTag">{tag}</Tag>)}
    </Card>
  )
}

Task.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  isDone: PropTypes.bool,
  tags: PropTypes.array.isRequired,
  onClickToggleDone: PropTypes.func,
  onClickDelete: PropTypes.func
}

export default Task;
