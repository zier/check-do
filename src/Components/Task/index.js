import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Card } from 'antd';

class Task extends Component {
  render() {
    const { title, description, onClick } = this.props

    return (
      <Card onClick={onClick} title={title} bordered={true}>
        {description}
      </Card>
    );
  }
}

Task.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  onClick: PropTypes.func
};

export default Task;
