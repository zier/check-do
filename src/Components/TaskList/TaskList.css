.taskListItem {
  margin-top: 10px;
}