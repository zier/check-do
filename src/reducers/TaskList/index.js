const TaskList = (state = {}, action) => {
  switch (action.type) {
    case 'ADD_TASK':
      return {
        displayItems: [
          ...state.displayItems,
          action.task
        ]
      }
    case 'REMOVE_TASK':
      const newList = state.TaskList.filter((task) => task.id !== action.id)

      return {
        ...state,
        TaskList: newList
      }
    default:
      return state
  }
}

export default TaskList
