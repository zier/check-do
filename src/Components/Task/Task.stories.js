import React from 'react';
import Task from '../Task';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

export const task = {
  title: 'Go home',
  description: 'go to home after meeting',
};

export const actions = {
  onClick: action('onClickTask'),
};


storiesOf('Task', module)
  .addDecorator(story => <div style={{ padding: '3rem' }}>{story()}</div>)
  .add('with data', () => <Task {...task} {...actions} />)
