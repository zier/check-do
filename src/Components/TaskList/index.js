import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Card, Col, Row } from 'antd';
import Task from '../Task';

const TaskList = ({ tasks, state, onSelectTask }) => {
  if (state === 'LOADING_TASKLIST') {
    return (
      <Row gutter={16}>
        <Col span={8}><Card loading title="Loading" /></Col>
      </Row>
    )
  }

  return (
    <Row gutter={16}>
      {
        tasks.map((task, index) => {
          return <Col span={8}><Task onClick={onSelectTask} {...task}  key={index}/></Col>
        })
      }
    </Row>
  )
}

TaskList.propTypes = {
  tasks: PropTypes.array,
  state: PropTypes.string,
  onSelectTask: PropTypes.func
};

export default TaskList;
