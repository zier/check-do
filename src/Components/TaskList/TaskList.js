import React from 'react'
import PropTypes from 'prop-types'

import { Card, Col, Row } from 'antd'
import Task from '../Task'
import './TaskList.css'

const TaskList = ({ tasks, state, onSelectTask }) => {
  if (state === 'LOADING_TASKLIST') {
    return (
      <Row className="taskList" gutter={16}>
        <Col span={8}><Card loading /></Col>
      </Row>
    )
  }

  return (
    <Row className="taskList" gutter={16}>
      {
        tasks.map((task, index) => {
          return <Col key={task.id} className="taskListItem" span={8}><Task onClick={onSelectTask} {...task}  key={index}/></Col>
        })
      }
    </Row>
  )
}

TaskList.propTypes = {
  tasks: PropTypes.array,
  state: PropTypes.string,
  onSelectTask: PropTypes.func
}

export default TaskList
