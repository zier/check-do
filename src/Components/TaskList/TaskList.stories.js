import React from 'react';
import TaskList from '../TaskList';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

export const emptyTask = []

export const tasks = [
  {
    title: 'Go home',
    description: 'go to home after meeting',
  },
  {
    title: 'Buy Eggs',
    description: 'go to supermarket',
  },
];

export const actions = {
  onSelectTask: action('onSelectTask'),
};

storiesOf('TaskList', module)
  .addDecorator(story => <div style={{ padding: '3rem' }}>{story()}</div>)
  .add('with data', () => <TaskList state={'LOADED_TASKLIST'} tasks={tasks} {...actions}/>)
  .add('with no data', () => <TaskList state={'LOADING_TASKLIST'} tasks={emptyTask} />)
