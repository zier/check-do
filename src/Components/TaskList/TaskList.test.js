import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TaskList from './TaskList.js'
import Task from '../Task/Task.js'

describe('TaskList work correctly', () => {
  const tasks = [
    {
      id: "1",
      title: 'Go home',
      description: 'go to home after meeting',
      tags: ["family", "urgent"],
    },
    {
      id: "2",
      title: 'Buy Eggs',
      description: 'go to supermarket',
      tags: ["family"],
    }
  ];

  // const actions = {
  //   onClickToggleDone: jest.fn(),
  //   onClickDelete: jest.fn(),
  // }

  const wrapper = shallow(<TaskList tasks={tasks} />)

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should have 2 task', () => {
    expect(wrapper.find(Task)).toHaveLength(2)
  })

})
