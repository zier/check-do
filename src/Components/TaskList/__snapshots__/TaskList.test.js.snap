// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList work correctly renders correctly 1`] = `
<Row
  className="taskList"
  gutter={16}
>
  <Col
    className="taskListItem"
    key="1"
    span={8}
  >
    <Task
      description="go to home after meeting"
      id="1"
      key="0"
      tags={
        Array [
          "family",
          "urgent",
        ]
      }
      title="Go home"
    />
  </Col>
  <Col
    className="taskListItem"
    key="2"
    span={8}
  >
    <Task
      description="go to supermarket"
      id="2"
      key="1"
      tags={
        Array [
          "family",
        ]
      }
      title="Buy Eggs"
    />
  </Col>
</Row>
`;
